crates/ruff_linter/resources/test/fixtures/pylint/unnecessary_list_index_lookup.py

```diff
@@ -0,0 +1,30 @@
+letters = ["a", "b", "c"]
+
+
+def fix_these():
+    [letters[index] for index, letter in enumerate(letters)]  # PLR1736
+    {letters[index] for index, letter in enumerate(letters)}  # PLR1736
+    {letter: letters[index] for index, letter in enumerate(letters)}  # PLR1736
+
+    for index, letter in enumerate(letters):
+        print(letters[index])  # PLR1736
+        blah = letters[index]  # PLR1736
+        assert letters[index]  == "d"  # PLR1736
+
+
+def dont_fix_these():
+    # once there is an assignment to the sequence[index], we stop emitting diagnostics
+    for index, letter in enumerate(letters):
+        letters[index] = "d"  # Ok
+        assert letters[index] == "d"  # Ok
```

Comment:
What about a case where `index` is mutated? e.g. `index += 1`

Reply:
Adding a case for that, as well as the sequence itself being mutated! Thanks!